oak-store-composite/src/main/java/org/apache/jackrabbit/oak/composite/MountInfoConfig.java

```diff
@@ -0,0 +1,115 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.composite;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.felix.scr.annotations.Activate;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.PropertyUnbounded;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.jackrabbit.oak.commons.PropertiesUtil;
+import org.osgi.framework.BundleContext;
+
+import static java.util.stream.Collectors.toList;
+
+@Component(label = "Apache Jackrabbit Oak MountInfo Config", configurationFactory = true, metatype = true)
+@Service(MountInfoConfig.class)
+public class MountInfoConfig {
+
+    private static final String[] PROP_MOUNTED_PATHS_DEFAULT = {};
+
+    @Property(label = "Mounted paths",
+        unbounded = PropertyUnbounded.ARRAY,
+        description = "Paths which are part of private mount"
+    )
+    private static final String PROP_MOUNT_PATHS = "mountedPaths";
+    private List<String> paths;
+
+    static final String PROP_MOUNT_NAME_DEFAULT = "private";
+
+    @Property(label = "Mount name",
+        description = "Name of the mount",
+        value = PROP_MOUNT_NAME_DEFAULT
+    )
+    private static final String PROP_MOUNT_NAME = "mountName";
+    private String mountName;
+
+    private static final String[] PROP_PATHS_SUPPORTING_FRAGMENTS_DEFAULT = {"/"};
+
+    @Property(label = "Paths supporting fragments",
+        unbounded = PropertyUnbounded.ARRAY,
+        description = "oak:mount-* under this paths will be included to mounts",
+        value = {"/"}
+    )
+    private static final String PROP_PATHS_SUPPORTING_FRAGMENTS = "pathsSupportingFragments";
+    private List<String> pathsSupportingFragments;
+
+    private static final boolean PROP_MOUNT_READONLY_DEFAULT = true;
+
+    @Property(label = "Readonly",
+        description = "If enabled then mount would be considered as readonly",
+        boolValue = PROP_MOUNT_READONLY_DEFAULT
+    )
+    private static final String PROP_MOUNT_READONLY = "readOnlyMount";
+
+    private boolean readOnly;
+
+    public MountInfoConfig() {
+    }
+
+    MountInfoConfig(String mountName, List<String> paths, List<String> pathsSupportingFragments, boolean readOnly) {
+        this.mountName = mountName;
+        this.paths = trimAll(paths);
+        this.pathsSupportingFragments = trimAll(pathsSupportingFragments);
+        this.readOnly = readOnly;
+    }
+
+    @Activate
+    private void activate(BundleContext bundleContext, Map<String, ?> config) {
```

Comment:
You should use component property types to ease dealing with defaults and generating meta type information at the same time.

Reply:
In the context of moving to the official OSGi component annotations, the properties are now defined on an `@interface` annotated with `@ObjectClassDefinition` and `@AttributeDefinition`.